src/bootstrap.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+use Silex\Application;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\HttpExceptionInterface;
+
+require_once __DIR__.'/../vendor/autoload.php';
+
+$app = new Application();
+
+$app->get('ping', function () {
+    return new Response(
+        'pong',
+        Response::HTTP_OK,
+        [
+            'Cache-Control' => 'must-revalidate, no-cache, no-store, private',
+            'Content-Type' => 'text/plain; charset=UTF-8',
+        ]
+    );
+});
+
+$app->get('/oauth2/authorize', function (Request $request) {
+    $redirectUri = $request->get('redirect_uri');
+    $state = $request->get('state');
+    $code = 'code_'.$state;
+    $location = $redirectUri.'?'.http_build_query([
+        'code' => $code,
+        'state' => $state,
+    ]);
+
+    return new Response(
+        'pong',
+        Response::HTTP_FOUND,
+        [
+            'Location' => $location,
+        ]
+    );
+});
+
+$app->post('/oauth2/token', function (Request $request) {
+    $code = $request->get('code');
+
+    return new JsonResponse([
+        'access_token' => 'access_token_'.$code,
+        'token_type' => 'bearer',
+        'refresh_token' => 'refresh_token_'.$code,
```

Comment:
We don't use this yet, but you'd end up with `refresh_token` rather than `code` in the input. (Should be replicate how ORCID works, or just the parts that we use?)

Reply:
The parts that we use, as soon as we use them - the API surface area is large (thinking about the errors for example)

Reply:
`refresh_token` should be removed.